x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/multi_line_string_diff_algorithm.test.ts

```diff
@@ -131,6 +131,31 @@ describe('multiLineStringDiffAlgorithm', () => {
         })
       );
     });
+
+    it('should handle analyzing long strings', () => {
```

Comment:
Would this unit test catch a performance issue like the one we experienced? Can we use a sufficiently large string payload to ensure the test times out if the performance issue reoccurs?

Reply:
This was used as a more realistic string field for testing, I can add another test that would fail based on string length. Might have to be an FTR test, I'll try both

Reply:
@xcrzx ok added an FTR test in https://github.com/elastic/kibana/pull/199388/commits/26fdf6e2b822cab65a2e93ddc6acbf06ffad1f7e since they actually timeout without any extra tooling needed. Put some results of the different regex parsers with the tests in the PR description.